source/grid/grid_tools.cc

```diff
@@ -5883,68 +5888,119 @@ namespace GridTools
-              const auto bb =
-                BoundingBox<spacedim>(entities[i]).create_extended(tolerance);
+      // loop over all entities
+      for (unsigned int i = 0; i < entities.size(); ++i)
+        {
+          // create a bounding box with tolerance
+          const auto bb =
+            BoundingBox<spacedim>(entities[i]).create_extended(tolerance);
 
-              // determine ranks potentially owning point/bounding box
-              std::set<unsigned int> my_ranks;
+          // determine ranks potentially owning point/bounding box
+          std::set<unsigned int> my_ranks;
 
-              for (const auto &box_and_rank :
-                   tree | boost::geometry::index::adaptors::queried(
-                            boost::geometry::index::intersects(bb)))
-                my_ranks.insert(box_and_rank.second);
+          for (const auto &box_and_rank :
+               tree | boost::geometry::index::adaptors::queried(
+                        boost::geometry::index::intersects(bb)))
+            my_ranks.insert(box_and_rank.second);
 
-              for (const auto rank : my_ranks)
-                ranks_and_indices.emplace_back(rank, i);
-            }
+          for (const auto rank : my_ranks)
+            ranks_and_indices.emplace_back(rank, i);
         }
-      else
+
+#else
+
+      auto fill_ranks_and_indices =
+        [&ranks_and_indices](
+          const std::vector<int> &                offsets,
+          const std::vector<std::pair<int, int>> &indices_ranks) {
+          for (unsigned int i = 0; i < offsets.size() - 1; ++i)
+            for (int j = offsets[i]; j < offsets[i + 1]; ++j)
+              ranks_and_indices.emplace_back(indices_ranks[j].second, i);
+        };
+
+      unsigned int idx = (global_bboxes.size() == 1) ?
+                           0 :
+                           Utilities::MPI::this_mpi_process(comm);
+      // size of global_bboxes>1
+      if constexpr (spacedim == 2)
+        {
+          // Create query bounding boxes out of entities
+          std::vector<BoundingBox<2>> query_bounding_boxes;
+          for (const auto &entity : entities)
+            query_bounding_boxes.push_back(
+              BoundingBox<2>(entity).create_extended(tolerance));
+          ArborXWrappers::BoundingBoxIntersectPredicate bb_intersect(
+            query_bounding_boxes);
+
+          ArborXWrappers::DistributedTree distributed_tree(comm,
+                                                           global_bboxes[idx]);
+
+          const auto &[indices_ranks, offsets] =
+            distributed_tree.query(bb_intersect);
+          fill_ranks_and_indices(offsets, indices_ranks);
+        }
+      else if constexpr (spacedim == 3)
         {
-          // TODO: use ArborX
+          ArborXWrappers::DistributedTree distributed_tree(comm,
+                                                           global_bboxes[idx]);
+
+          if constexpr (std::is_same<T, Point<3>>::value)
```

Comment:
Any reason why you didn't implemented `Point` for the 2D case?

Reply:
The reason is that the `get()` method was returning a `Point<3>`, so I wasn't sure what was the best option.